cheroot/workers/threadpool.py

```diff
@@ -151,12 +151,20 @@ def __init__(
             server (cheroot.server.HTTPServer): web server object
                 receiving this request
             min (int): minimum number of worker threads
-            max (int): maximum number of worker threads
+            max (int): maximum number of worker threads (-1 for no max)
             accepted_queue_size (int): maximum number of active
                 requests in queue
             accepted_queue_timeout (int): timeout for putting request
                 into queue
+
+        Raises:
+            ValueError: if the min/max values are invalid
         """
+        if min < 1:
+            raise ValueError('min must be > 0')
+        if -1 < max < min:
```

Comment:
I've been staring at this for a while and realized that the `-1` value is just a dirty hack. It would be more fitting for the low-level ecosystems like C as a magical constant.
In Python, though, we can use the actual infinity, which is semantically closer to what it attempts to represent.
For compatibility, though, we could internally convert `-1` into infinity.
```suggestion
        if max < 0:
            max = float('inf')
        if max < min:
```
(`float('inf') is math.inf`)